core/src/main/java/org/mapfish/print/http/MfClientHttpRequestFactoryImpl.java

```diff
@@ -113,7 +113,7 @@ private static final class RandomizingDnsResolver extends SystemDefaultDnsResolv
     public InetAddress[] resolve(final String host) throws UnknownHostException {
       final List<InetAddress> list = Arrays.asList(super.resolve(host));
       Collections.shuffle(list);
-      return list.toArray(new InetAddress[list.size()]);
+      return list.toArray(new InetAddress[0]);
```

Comment:
Why this change?

Reply:
Qodana reported:
n older Java versions, using a pre-sized array was recommended, as the reflection call necessary to create an array of proper size was quite slow.
However, since late updates of OpenJDK 6, this call was intrinsified, making the performance of the empty array version the same, and sometimes even better, compared to the pre-sized version. Also, passing a pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray calls. This may result in extra nulls at the end of the array if the collection was concurrently shrunk during the operation.